File: ifoodAndroidTest/app/src/main/java/test/ifood/uellisson/ifoodandroidtest/view/TweetViewHolder.java
package test.ifood.uellisson.ifoodandroidtest.view;

import android.support.annotation.NonNull;
import android.support.v7.widget.RecyclerView;
import android.view.View;
import android.widget.TextView;

import butterknife.Bind;
import butterknife.ButterKnife;
import test.ifood.uellisson.ifoodandroidtest.R;

public class TweetViewHolder extends RecyclerView.ViewHolder {

    @Bind(R.id.tweet_message)
    TextView tweetMessage;

    @Bind(R.id.create_at)
    TextView crateAt;

    public TweetViewHolder(@NonNull View itemView) {
        super(itemView);
        ButterKnife.bind(this, itemView);
    }
}

File: app/src/main/java/br/com/andreyneto/ifood_mobile_test/ui/main/TweetViewHolder.kt
package br.com.andreyneto.ifood_mobile_test.ui.main

import android.view.View
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import br.com.andreyneto.ifood_mobile_test.R
import br.com.andreyneto.ifood_mobile_test.data.database.TweetEntry
import kotlinx.android.synthetic.main.tweet_item.view.*

class TweetViewHolder(itemView: View):
        RecyclerView.ViewHolder(itemView) {
    fun bind(tweet: TweetEntry, clickListener: (Long) -> Unit) {
        itemView.textView.text = tweet.text
        itemView.view.visibility = if (tweet.sentimentChecked) View.VISIBLE else View.GONE
        when {
            tweet.score < -0.25 -> {
                itemView.view.setBackgroundColor(ContextCompat.getColor(itemView.context!!, R.color.colorSad))
            }
            tweet.score > 0.25 -> {
                itemView.view.setBackgroundColor(ContextCompat.getColor(itemView.context!!, R.color.colorHappy))
            }
            else -> {
                itemView.view.setBackgroundColor(ContextCompat.getColor(itemView.context!!, R.color.colorNeutral))
            }
        }
        itemView.setOnClickListener { clickListener(tweet.tweetID)}
    }
}